### ISSUE TITLE
How to set up replication?

### ISSUE THREAD

**Issue by volkanunsal** [docker-library/postgres#55]
I have been trying to set up replication on the version 9.3 of the image, and having quite a lot of difficulty. I'm curious if this is possible... The README states the "recent versions" are able to support replication, but is 9.3 one of those?

**Comment by stapelberg** [comment]
@volkanunsal Yes, 9.3 supports replication: http://www.postgresql.org/docs/9.3/static/runtime-config-replication.html

If you have any further questions, I think it’d be better if you asked in the corresponding PostgreSQL support issues. This repository is for specifics of the docker image, not PostgreSQL itself.

**Comment by volkanunsal** [comment]
I've had a long back and forth on the Postgres support forums about this. I have eliminated all other possibilities, and exhausted the patience of the mailing list before posting here. The gist of the discussion can be boiled down to [this gist.](https://gist.github.com/volkanunsal/ad2173e2649393fcd3b6) You will find the logs from both the server and the client on that gist. The problem is that the client does not start the `walreceiver` service after it's booted up when it's installed using a Docker image. When I installed Postgres to the server directly, it doesn't have this problem. 

I'm not a Postgres expert, nor a Docker expert. Hopefully, someone can look at this and point out what I did wrong. If there are any working examples of a replication setup out in the wild that you can point me to, that would be much appreciated as well.

**Comment by stapelberg** [comment]
You should have started this issue with the data you just provided here :). Can you also add a link to the discussion you’ve had on the Postgres forum so that we know what you’ve tried and what you haven’t?

I also want to set up PostgreSQL replication using docker, and will likely face the same problem then. I’m not sure when I’ll get around to it, though.

**Comment by volkanunsal** [comment]
[Here is the discussion.](http://www.postgresql.org/search/?m=1&l=2&q=no+pg_hba.conf+entry+for+replication+connection+from+host) It's not threaded and might be hard to follow because it's just a Listserv.

**Comment by pikeas** [comment]
@volkanunsal Were you able to get PG replication working between containers?

**Comment by volkanunsal** [comment]
Nope.

**Comment by tianon** [comment]
Looking at Google, it would appear that others have figured out how to accomplish this: https://www.google.com/search?q=docker+postgres+replication :smile:

For example, https://github.com/DanielDent/docker-postgres-replication / https://hub.docker.com/r/danieldent/postgres-replication/

**Comment by tianon** [comment]
Apparently past @tianon (https://github.com/docker-library/postgres/issues/104#issuecomment-173980602) actually had it working too, and present @tianon just doesn't remember:

> The Postgres replication is a little bit complex, but this is definitely doable.  I went through some of https://www.digitalocean.com/community/tutorials/how-to-set-up-master-slave-replication-on-postgresql-on-an-ubuntu-12-04-vps to get something going and it's not too bad.  The user created by our entrypoint script is a `SUPERUSER`, so it does have permission to replicate by default, so I think the most difficult parts here are setting the config (`wal_level`, etc) and doing the initial replication.

**Comment by wglambert** [comment]
Since this issue seems unrelated to any errors in the image itself, and being two years without further comment, I'm going to prune the issue.
If you believe this to be in error then let me know and I'll re-open it

Given that the issue is more of a usability question, it would be more suited for [the Docker Community Forums](https://forums.docker.com/), [the Docker Community Slack](https://blog.docker.com/2016/11/introducing-docker-community-directory-docker-community-slack/), or [Stack Overflow](https://stackoverflow.com/search?tab=newest&q=docker).
